Work/nested serialization

jvbsl opened the pull request.

jvbsl committed:
Implemented nested serialization feature
* Implemented serialization completions to be able to continue or retry serialization chain
* Made SerializerMask parameter for serializers ref parameters to be able to manipulate them for subsequent serializers
* Implemented serializer and deserializer priority seperately to be able to have different order of serialization than deserialization(can help in impproving performance for nested serializers)

jvbsl committed:
Implemented TypeConverter attribute used to inject a intermediate type for serialization
* Added NoosonConversionAttribute to mark a property to use an intermediate type(containing implicit conversion operators from the property type), or a INoosonConverter implementation with an optional type to ConvertTo(only needed for ambiguity resolvement when multiple converters are implemented)
* Added INoosonConverter interface for implementation of a converter to convert from a type into an intermediate type
* Implemented Finalizer serializers, which are executed after a successfull serialization chain(needed for the type converter serialization, as the conversion at the deserialization needs to happen after all other deserializers have completed)
* Made property parameter ref for manipulation of type information for subsequent serializers

susch19 commented on NonSucking.Framework.Serialization.SerializerCollector/SerializeGeneratorCollector.cs:
```diff
@@ -66,9 +67,38 @@ private static int GetPriority(Compilation compilation, ClassDeclarationSyntax c
+                        return new(serializerPriority, serializerPriority, isFinalizer);
+                    }
+
+                    if (list is { Count: 1 })
```
```suggestion
                    if (list is { Count: 2 })
```

jvbsl committed:
Update NonSucking.Framework.Serialization.SerializerCollector/SerializeGeneratorCollector.cs

Actually use correct number of arguments for StaticSerializer attribute interpretation

Co-authored-by: Sascha <suschi@ist-einmalig.de>
